patches/tModLoader/Terraria/ModLoader/IO/TagIO.cs

```diff
@@ -118,7 +122,7 @@ private class ClassPayloadHandler<T> : PayloadHandler<T> where T : class
 						return GetHandler(GetPayloadId(v.GetType().GetGenericArguments()[0])).CloneList(v);
 					}
 					catch (IOException) {
-						throw new ArgumentException("Invalid NBT list type: "+v.GetType());
```

Comment:
Is there some good reason this was an `ArgumentException` instead of an `IOException`?